pkgs/tools/typesetting/tex/texlive/default.nix

```diff
@@ -154,15 +141,36 @@ in
             maintainers = with lib.maintainers;  [ veprbl ];
             license = licenses.${pname};
           }
-          (combine {
+          (self.combine {
             ${pname} = attrs;
             extraName = "combined" + lib.removePrefix "scheme" pname;
             extraVersion = with version; if final then "-final" else ".${year}${month}${day}";
           })
         )
-        { inherit (tl)
+        { inherit (self.pkgs)
             scheme-basic scheme-context scheme-full scheme-gust scheme-infraonly
             scheme-medium scheme-minimal scheme-small scheme-tetex;
         }
     );
-  }
+  };
+
+  texlive = makeScopeWithSplicing' {
+    otherSplices = generateSplicesForMkScope "texlive";
+    f = addPackages;
+  };
+
+  applyOverScope = f: scope: f (scope // {
+      overrideScope = g: applyOverScope f (scope.overrideScope g);
+  });
+
+  # for backward compability
+  compatFixups = scope:
+    # TODO
+    scope.pkgs // # remove this line to fix cross
+    scope // {
+      bin = scope.bin // {
+        latexindent = lib.findFirst (p: p.tlType == "bin") scope.pkgs.latexindent.pkgs;
+      };
+    };
+
+in applyOverScope compatFixups texlive
```

Comment:
Now that I understand what is going on, I managed to verify that
```nix
lib.last pkgsCross.aarch64-multiplatform.__splicedPackages.texlive.pkgs.a2ping.pkgs
```
does not have its own splices. I assume that the binary containers need to be spliced as well, don't they? So we really need the multi-output to make this work properly.

Reply:
Yes, ideally, the bin containers should also be spliced. This isn't a huge deal yet as there isn't anything cross-compilable that uses the bin containers yet and I doubt users will appear while the interface is this terrible and they aren't really exposed.

But this is probably™ something that fixes itself once the better format for the containers lands.